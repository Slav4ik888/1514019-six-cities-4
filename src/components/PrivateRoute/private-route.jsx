import React from 'react';
import pt from 'prop-types';

import {connect} from 'react-redux';
import {Route, Redirect} from 'react-router-dom';

import {AuthStatus} from '../../reducers/user/user.js';
import {getUserStatus} from '../../reducers/user/selectors.js';

import {AppRoute} from '../../utils/const.js';


const PrivateRoute = ({exact, path, render, userStatus}) => {
  // console.log('PRIV userStatus: ', userStatus);

  return (
    <Route
      exact={exact}
      path={path}
      render={() => {
        return (
          userStatus === AuthStatus.AUTH
            ? render()
            : <Redirect to={AppRoute.LOGIN}/>
        );
      }}
    />
  );
};

PrivateRoute.propTypes = {
  exact: pt.bool.isRequired,
  path: pt.string.isRequired,
  render: pt.func.isRequired,
  userStatus: pt.string.isRequired
};

const mapStateToProps = (state) => ({
  userStatus: getUserStatus(state),
});

export {PrivateRoute};
export default connect(mapStateToProps, undefined)(PrivateRoute);
