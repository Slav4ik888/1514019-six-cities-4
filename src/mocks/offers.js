export default [
  {
    id: 1,
    isPremium: true,
    isFavourite: false,
    previewImage: `img/apartment-03.jpg`,
    price: 180,
    rating: 5,
    cardTitle: `Beautiful & luxurious apartment at great location`,
    offerType: `House`,
  },
  {
    id: 2,
    isPremium: true,
    isFavourite: false,
    previewImage: `img/apartment-01.jpg`,
    price: 180,
    rating: 3,
    cardTitle: `Nice, cozy, warm big bed apartment`,
    offerType: `Hotel`
  },
  {
    id: 3,
    isPremium: false,
    isFavourite: false,
    previewImage: `img/apartment-02.jpg`,
    price: 10,
    rating: 5,
    cardTitle: `Wood and stone place`,
    offerType: `Apartment`,
  },
  {
    id: 4,
    isPremium: false,
    isFavourite: false,
    previewImage: `img/apartment-01.jpg`,
    price: 180,
    rating: 5,
    cardTitle: `Beautiful & luxurious apartment at great location`,
    offerType: `Private room`,
  },
  {
    id: 5,
    isPremium: false,
    isFavourite: true,
    previewImage: `img/apartment-03.jpg`,
    price: 480,
    rating: 4,
    cardTitle: `Beautiful & luxurious apartment at great location`,
    offerType: `Hotel`
  },
  {
    id: 6,
    isPremium: false,
    isFavourite: false,
    previewImage: `img/apartment-01.jpg`,
    price: 680,
    rating: 3,
    cardTitle: `Nice, cozy, warm big bed apartment`,
    offerType: `House`,
  },
  {
    id: 7,
    isPremium: false,
    isFavourite: true,
    previewImage: `img/apartment-02.jpg`,
    price: 180,
    rating: 4,
    cardTitle: `Wood and stone place`,
    offerType: `Apartment`,
  },
  {
    id: 8,
    isPremium: true,
    isFavourite: false,
    previewImage: `img/apartment-01.jpg`,
    price: 180,
    rating: 4,
    cardTitle: `Beautiful & luxurious apartment at great location`,
    offerType: `Private room`,
  },
];
