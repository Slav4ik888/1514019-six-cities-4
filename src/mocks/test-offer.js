export const testOffer = {
  id: 1,
  isPremium: true,
  isFavourite: false,
  previewImage: `img/apartment-03.jpg`,
  pictures: [
    `img/apartment-01.jpg`,
    `img/apartment-02.jpg`,
    `img/apartment-03.jpg`,
    `img/room.jpg`,
    `img/studio-01.jpg`,
    `img/studio-photos.jpg`],
  amenities: [`Wifi`, `Heating`, `Kitchen`, `Cable TV`],
  bedrooms: 4,
  maxGuestsNumber: 5,
  description: `This cozy and complete apartment in the heart of New York is a typical, traditional and authentic feel for how New Yorkers live. I welcome all guests to stay at this place during their stay and visit of New York. 
  Please note that this apartment is on the sixth floor and there is no lift (elevator).
  The apartment is convenient for all guests, fitted with a kitchen, small dining area and two bedrooms – couples and groups of friends will find this accommodation most optimal.`,
  host: {
    photo: `img/avatar-angelina.jpg`,
    name: `Angelina Jonson`,
    super: true
  },
  price: 180,
  rating: 5,
  cardTitle: `Beautiful & luxurious apartment at great location`,
  offerType: `House`,
  coordinates: [48.856663, 2.351556],
  city: {
    zoom: 8,
    coordinates: [48.856663, 2.351556],
  },
  location: {
    zoom: 8,
  },
  reviews: [
    {
      id: 1,
      author: {
        photo: `img/avatar-angelina.jpg`,
        name: `Strange jober`
      },
      description: `small dining area and two bedrooms – couples and groups of friends will find this accommodation`,
      date: 1593095836000,
      rating: 3
    },
    {
      id: 2,
      author: {
        photo: ``,
        name: `Slava`
      },
      description: `dining area and five bedrooms – couples and groups of friends will find this accommodation`,
      date: 1593095936000,
      rating: 4.7
    },
  ],

};
