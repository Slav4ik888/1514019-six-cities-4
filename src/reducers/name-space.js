export const NameSpace = {
  DATA: `DATA`,
  TRAVEL: `TRAVEL`,
  USER: `USER`,
  FAVORITE: `FAVORITE`,
};
