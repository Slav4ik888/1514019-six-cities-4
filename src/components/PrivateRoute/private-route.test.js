import React from 'react';
import renderer from 'react-test-renderer';
import {Provider} from 'react-redux';
import configureStore from 'redux-mock-store';
import {BrowserRouter} from 'react-router-dom';

import {PrivateRoute} from './private-route.jsx';

import {NameSpace} from '../../reducers/name-space.js';
import {offers} from '../../mocks/offers.js';
import {testOffer, reviews, nearbyOffers} from '../../mocks/test-offer.js';


const mockStore = configureStore([]);

const authInfo = {
  avatarUrl: `/static/avatar/7.jpg`,
  email: `korzan.va@mail.ru`,
  id: 1,
  isPro: false,
  name: `Vyacheslav`,
};


describe(`Snapshot <PrivateRoute/> correctly`, () => {
  it(`Render <PrivateRoute/>  'AUTH'`, () => {
    const store = mockStore({
      [NameSpace.TRAVEL]: {
        activeCity: 0,
        activeOffer: testOffer,
        activeHoverOffer: null,
      },
      [NameSpace.USER]: {
        userStatus: `AUTH`,
        authInfo,
        isLoading: false,
      },
      [NameSpace.DATA]: {
        allOffers: offers,
        comments: reviews,
        nearbyOffers,
        isLoading: false,
      },
    });

    const tree = renderer.create(
        <Provider store={store}>
          <BrowserRouter>
            <PrivateRoute
              exact={true}
              path={`/favorites`}
              render={() => {
                return <div/>;
              }}
              userStatus={`AUTH`}
            />
          </BrowserRouter>
        </Provider>
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it(`Render <PrivateRoute/>  'NO_AUTH'`, () => {
    const store = mockStore({
      [NameSpace.TRAVEL]: {
        activeCity: 0,
        activeOffer: testOffer,
        activeHoverOffer: null,
      },
      [NameSpace.USER]: {
        userStatus: `NO_AUTH`,
        authInfo,
        isLoading: false,
      },
      [NameSpace.DATA]: {
        allOffers: offers,
        comments: reviews,
        nearbyOffers,
        isLoading: false,
      },
    });

    const tree = renderer.create(
        <Provider store={store}>
          <BrowserRouter>
            <PrivateRoute
              exact={true}
              path={`/favorites`}
              render={() => {
                return <div/>;
              }}
              userStatus={`NO_AUTH`}
            />
          </BrowserRouter>
        </Provider>
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

});
