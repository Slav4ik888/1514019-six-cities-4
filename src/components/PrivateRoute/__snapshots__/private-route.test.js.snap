// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot <PrivateRoute/> correctly Render <PrivateRoute/>  'AUTH' 1`] = `null`;

exports[`Snapshot <PrivateRoute/> correctly Render <PrivateRoute/>  'NO_AUTH' 1`] = `null`;
